draft-ietf-quic-http.md

```diff
@@ -549,20 +549,38 @@ SETTINGS frame MUST be processed in the order they appear, with no other frame
 processing between values. Unsupported parameters MUST be ignored. 
 
 Once all values have been processed, if the REQUEST_ACK flag was set, the 
-recipient MUST immediately emit a SETTINGS_ACK frame listing the identifiers 
-whose values were understood and applied. (If none of the values were 
-understood, the SETTINGS_ACK frame will be empty, but MUST still be sent.) Upon 
-receiving an SETTINGS_ACK frame, the sender of the altered parameters can rely 
-on the setting having been applied. 
-
-TODO:
-: The above text was written for HTTP/2 -- QUIC has cross-stream timing issues 
-here that need to be solved.
-
+recipient MUST immediately emit the following:
+
+ - On the connection control stream, a SETTINGS_ACK frame listing the 
+   identifiers whose values were understood and applied. (If none of the values 
+   were understood, the SETTINGS_ACK frame will be empty, but MUST still be
+   sent.) 
+
+ - On each request control stream which is not in the "half-closed (local)" or
+   "closed" state, an empty SETTINGS_ACK frame.
+
+The SETTINGS_ACK frame on the connection control stream contains the highest
+stream number which was open at the time the SETTINGS frame was received.  All
+streams with higher numbers can safely be assumed to have the new settings in
+effect when they open.
```

Comment:
The first case is the corner-case from line 572 -- because QUIC mandates that streams be opened "sequentially," stream-skipping is already prohibited.  (Should that be made clearer?)  The fact that I've seen stream N opened means N-2 is in-flight, even if I haven't seen it yet.  That means, having received a SETTINGS frame, I can go ahead and send a SETTINGS_ACK on N-2 *without having yet received a frame* because it's not out-of-order at the remote side.

Your suggested correlation text looks a lot like what I have at line 576 about what's needed to consider the ACK "finished."  I might need to add a note that all previous SETTINGS frames have to be "finished" before a subsequent one can be -- ACKs on-stream are fully ordered.

I don't see the issue when the receiver of SETTINGS opens the stream.  There are two possible orderings:

- **SETTINGS processed before stream created.**  Then the SETTINGS_ACK on stream 3 will not cover the new stream, and the stream abides by the new settings from the start.
- **Stream created before SETTINGS processed.**  Then the SETTINGS_ACK on stream 3 *will* cover the new stream, and the sender will expect a STREAM_ACK (which may be the first frame on the stream).

What am I missing?